### ISSUE TITLE
[Testerina] Error while trying to mock jdbc:Client 

### ISSUE THREAD

**Issue by Dilhasha** [ballerina-platform/ballerina-lang#25441]
**Description:**

Throws following error during object mocking jdbc:Client.
TypeCastError message=incompatible types: 'error' cannot be cast to 'test:InvalidObjectError|test:FunctionNotFoundError|test:FunctionSignatureMismatchError|test:InvalidMemberFieldError|test:FunctionCallError|()'
Identified during https://github.com/ballerina-platform/ballerina-lang/issues/25280

**Steps to reproduce:**
Try this example[1] and to the test file, add the following beforeSuite function.

```
@test:BeforeSuite
function beforeSuiteFunc() {
    testDB = <jdbc:Client>(test:mock(jdbc:Client));
    test:prepare(testDB).when("update").doNothing();
    io:println("I'm the before suite function!");
}
```

When building the module, you will get the following error.

```
        [fail] mockRegistryDBConnection [before test suite function]:
            {ballerina}TypeCastError message=incompatible types: 'error' cannot be cast to 'test:InvalidObjectError|test:FunctionNotFoundError|test:FunctionSignatureMismatchError|test:InvalidMemberFieldError|test:FunctionCallError|()'
        at ballerina.test:thenReturnExt(mock.bal:364)
           ballerina.test.MemberFunctionStub:doNothing(mock.bal:197)
```

[1] https://ballerina.io/learn/by-example/jdbc-client-crud-operations.html

**Affected Versions:**
Ballerina 1.2.7

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by azinneera** [comment]
The reason for resulting in an exception is that the `jdbc:Client->update` function has the return type as `jdbc:UpdateResult|error`. `doNothing()` can only be used when the function has no return type or has an optional return type. Nonetheless, there is a bug in casting the error message which needs to be fixed.

**Comment by Dilhasha** [comment]
@azinneera Thanks, you are correct. I was able to fix it by returning jdbc:UpdateResult. 
Yes, we need to fix that misleading error message.

**Comment by dilee** [comment]
I’ve encountered the same issue when mocking a JDBC client when mocking a **_select_** operation. As mentioned in the issue, I’ve set the proper return type (in this case, _table<record {}>|Error_ : https://github.com/ballerinax/jdbc/blob/7dc579f049ab319cf10a5d9507a981112d989b14/jdbc-extension/src/main/ballerina/ballerinax/jdbc/endpoint.bal#L67) but I’m still getting the same error.

**Comment by Dilhasha** [comment]
The issue occurs when trying to mock jdbc:Client using 'withArguments' as follows.

`test:prepare(dbClient).when("select").withArguments("SELECT name, age FROM settings WHERE user_id='test 1'", test:ANY).thenReturn(TableRowResponse());`

But, this works when trying to mock an http:Client.

As a temporary workaround, let's avoid using the 'withArguments' and handle any consequent cases using https://ballerina.io/learn/testing-ballerina-code/mocking/#stubbing-with-multiple-values-to-return-sequentially-for-each-function-call

Meanwhile we will get this fixed for ballerina 1.2.x

**Comment by Dilhasha** [comment]
> The issue occurs when trying to mock jdbc:Client using 'withArguments' as follows.
> 
> `test:prepare(dbClient).when("select").withArguments("SELECT name, age FROM settings WHERE user_id='test 1'", test:ANY).thenReturn(TableRowResponse());`
> 
> But, this works when trying to mock an http:Client.
> 
> As a temporary workaround, let's avoid using the 'withArguments' and handle any consequent cases using https://ballerina.io/learn/testing-ballerina-code/mocking/#stubbing-with-multiple-values-to-return-sequentially-for-each-function-call
> 
> Meanwhile we will get this fixed for ballerina 1.2.x

This issue is fixed with #26656 

We will also update the error handling to give meaningful errors.

**Comment by Dilhasha** [comment]
@dilee All the issues related to mocking are fixed and will be available with the next 1.2.x release.
